examples/browser/test/pages/main-page.ts

```diff
@@ -57,15 +63,102 @@ export class MainPage {
         }
     }
 
+    /* Returns true id Menu is loaded */
     public waitForLoadingMenu(): void {
         browser.waitForExist('.p-Widget.p-Menu.p-MenuBar-menu');
     }
 
+    /* Get the Location of the Tab */
     public getxBarTabPosition(tabNumber: Number): Number {
-        return browser.getLocation(`ul.p-MenuBar-content > .p-MenuBar-item:nth-child(${tabNumber}`, 'x')
+        return browser.getLocation(`ul.p-MenuBar-content > .p-MenuBar-item:nth-child(${tabNumber}`, 'x');
     }
 
+    /* Right click on directory structure to display the context menu works */
     public getxMenuPosition(): Number {
-        return this.driver.getLocation('.p-Widget.p-Menu.p-MenuBar-menu', 'x')
+        return this.driver.getLocation('.p-Widget.p-Menu.p-MenuBar-menu', 'x');
+    }
+
+    /* Test to check if the browser directory expands to shows files within it */
+    public isFolderStructureExpandable(): Boolean {
+        browser.waitForExist('.theia-TreeNodeCaption');
+        return browser.isExisting('span.theia-ExpansionToggle');
+    }
+
+    /* Right click on directory structure to display the context menu works */
+    public isRightClickDirectoryAvailable(): Boolean {
+        browser.click('li.p-TabBar-tab');
+        browser.rightClick('#files.p-Widget.theia-Tree.theia-FileTree.theia-Files.p-StackedPanel-child', 50, 50);
+        return browser.isVisible('.p-Widget.p-Menu');
+    }
+
+    /* Right Click on Theia Main Panel to display  context menu */
+    public isRightClickInMainPanelAvailable(): Boolean {
+        browser.rightClick('#theia-main-dock-panel');
+        return browser.isVisible('.p-Widget.p-Menu');
+    }
+
+    /* Double Click on the file to open */
+    public isDoubleClickOnFileAvailable(): Boolean {
+        browser.waitForExist('.theia-TreeNodeCaption');
+        browser.doubleClick('.theia-TreeNodeCaption');
+        return browser.isVisible('#theia-left-stack.p-Widget.p-Panel.p-StackedPanel.p-SplitPanel-child');
     }
+
+    /* Click on OPEN in the context menu to open the file in the workspace */
+    public isContextMenuOpenAvailable(filePositionNumber: Number): Boolean {
```

Comment:
Is there a chance to be able to lookup items by their names? That would be awesome. Otherwise, it could break easily.

Reply:
There is. You can lookup elements by their name or id. But not all our elements have a name and id associated with them.

Reply:
Great! Thanks for the clarification. I must have overlooked that.